app/src/main/java/com/eventyay/organizer/data/role/RoleApi.java

```diff
@@ -20,4 +21,10 @@
 
     @DELETE("role-invites/{role_invite_id}")
     Completable deleteRole(@Path("role_invite_id") long roleInviteId);
+
+    @PATCH("role-invites/{role_invite_id}")
+    Observable<RoleInvite> updateRole(@Path("role_invite_id") long roleInviteId,@Body RoleInvite roleInvite);
```

Comment:
As discussed, this is wrong

Reply:
Actually, when creating a role we send role_invite with setting role with it.
and role only contains spinnerSelectedItem+1.
So we can't change role. Since role doesn't contain anything other than spinnerSelectedItem+1.

In order to add an option to edit roles, we need to use role_invite. 


RoleInviteFragment.java
> roleId = binding.selectRole.getSelectedItemPosition() + 1;
            roleInviteViewModel.createRoleInvite(roleId);

RoleInviteViewModel.java
` public void createRoleInvite(long roleId) {
        long eventId = ContextManager.getSelectedEvent().getId();
        Event event = new Event();
        event.setId(eventId);
        roleInvite.setEvent(event);
        role.setId(roleId);
        roleInvite.setRole(role);

        compositeDisposable.add(
            roleRepository
                .sendRoleInvite(roleInvite)
                .doOnSubscribe(disposable -> progress.setValue(true))
                .doFinally(() -> progress.setValue(false))
                .subscribe(sentRoleInvite -> {
                    success.setValue("Role Invite Sent");
                    dismiss.call();
                }, throwable -> error.setValue(ErrorUtils.getMessage(throwable).toString())));
    }`

Reply:
I already told you about 5 times why it won't work. I won't waste any more time on this. Sorry